File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def __init__(self):
        self.dic = {1:1, 2:2}
    
    def climbStairs(self, n):
        if n not in self.dic:
            self.dic[n] = self.climbStairs(n-1) + self.climbStairs(n-2)
        return self.dic[n]

File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {        
        vector<int> ans(n, 0);
        if(n==1){
            return 1;
        }
        if(n==2){
            return 2;
        }
        ans[0] = 1;
        ans[1] = 2;
        
        // recusrive call
        for(int i=2; i<n; i++){
            ans[i] = ans[i-1] + ans[i-2];
        }
        
        return ans[n-1];
    }
};